config/express.js

```diff
@@ -1,91 +1,88 @@
 /**
  * Module dependencies.
  */
-var express = require('express'),
-    mongoStore = require('connect-mongo')(express),
-    flash = require('connect-flash'),
-    helpers = require('view-helpers'),
-    config = require('./config');
-
-module.exports = function(app, passport, mongoose) {
-    app.set('showStackError', true);
-
-    //Should be placed before express.static
-    app.use(express.compress({
-        filter: function(req, res) {
-            return (/json|text|javascript|css/).test(res.getHeader('Content-Type'));
-        },
-        level: 9
+import express from 'express';
+import connect from 'connect-mongo';
+import flash from 'connect-flash';
+import helpers from 'view-helpers';
+import config from './config';
+
+const mongoStore = connect(express);
+
+export default (app, passport, mongoose) => {
+  app.set('showStackError', true);
+
+  // Should be placed before express.static
+  app.use(express.compress({
+    filter: (req, res) =>
+      (/json|text|javascript|css/).test(res.getHeader('Content-Type')),
+    level: 9
+  }));
+
+  // Setting the fav icon and static folder
+  app.use(express.favicon());
+  app.use(express.static(config.root + '/public'));
+
+  // Don't use logger for test env
+  if (process.env.NODE_ENV !== 'test') {
+    app.use(express.logger('dev'));
+  }
+
+  // Set views path, template engine and default layout
+  app.set('views', config.root + '/app/views');
+  app.set('view engine', 'jade');
+
+  // Enable jsonp
+  app.enable('jsonp callback');
+
+  app.configure(() => {
+    // cookieParser should be above session
+    app.use(express.cookieParser());
+
+    // bodyParser should be above methodOverride
+    app.use(express.bodyParser());
+    app.use(express.methodOverride());
+
+    // express/mongo session storage
+    app.use(express.session({
+      secret: 'MEAN',
```

Comment:
we should totally avoid directory including application settings or secret keys in the codebase